### PR TITLE
use six.text_type(exception) instead of exception.message

### PR DESCRIPTION
**Pull request by nickpell**

### CODE REVIEW

File: corehq/ex-submodules/dimagi/utils/logging.py [hunk 1]
```diff
@@ -23,7 +23,7 @@ def notify_exception(request, message=None, details=None, exec_info=None):
     if request is not None:
         message = message or request.path
     if isinstance(message, bytes):
-        message = message.encode('utf-8')
+        message = message.decode('utf-8')
```
Comment (millerdev) [inline corehq/ex-submodules/dimagi/utils/logging.py]:
```suggestion
        try:
            message = message.decode('utf-8')
        except UnicodeDecodeError:
            message = repr(message)
```

**Review by millerdev — APPROVED** [review]

File: corehq/ex-submodules/dimagi/utils/logging.py (hunk 1 above)
Comment (nickpell) [reply corehq/ex-submodules/dimagi/utils/logging.py]: Is the concern here that the bytes might not be encoded in UTF-8, or something separate?

Comment (millerdev) [reply corehq/ex-submodules/dimagi/utils/logging.py]: correct

Comment (nickpell) [reply corehq/ex-submodules/dimagi/utils/logging.py]: Hmm ... Wouldn't it be preferable to standardize on only using UTF-8?  I don't think I've encountered any exceptions that use a non-UTF-8 encoding in HQ before.

Also fwiw after the Python migration I'd support curtailing our uses of ```isinstance(..., bytes)``` and assuming/asserting that string-like objects are ```str``` (```unicode```) to enforce proper Unicode Sandwiching.

Comment (millerdev) [reply corehq/ex-submodules/dimagi/utils/logging.py]: My concern is more about improper use of `notify_exception(request, message, ...)` where a non-text or badly encoded bytes object is (probably mistakenly) passed in as the message argument, causing a decoding error, and preventing the original exception from being reported to sentry.

In general, I support your proposal to assume string-like objects are text. However, in this case I think extra care is warranted to hopefully get the error sent through to sentry even if the message is malformed.

Comment (nickpell) [reply corehq/ex-submodules/dimagi/utils/logging.py]: https://github.com/dimagi/commcare-hq/pull/24355

**Pull request merged**